### ISSUE TITLE
Usernames with dots in them break routes

### ISSUE THREAD

**Issue by Gargron** [tootsuite/mastodon#6860]
See https://github.com/tootsuite/mastodon/pull/6844#issuecomment-374993289

This is because the dot is used as separator for format in Rails. Before you could easily pull up the ActivityPub JSON of a resource by adding `.json` to the URL, if we change the constraints of the routes, this will become impossible.

**Comment by Gargron** [comment]
Decided not to whitelist dots.

**Comment by beedell-roke** [comment]
@Gargron, what occurs to usernames registered with periods?

**Comment by tigre-bleu** [comment]
I don't know if that answers the question but when I tried long ago to deploy Mastodon using my LDAP as authentication backend, it was not possible because usernames were the email user part (ie john.doe for my user john.doe@example.com).

See the attached issue I wrote: https://github.com/mastodon/mastodon/issues/6830

So it was just plain impossible to deploy Mastodon in my infrastructure and I gave up. From what I understand the problem is rooted low in the fedi architecture and it was decided not to change it.
